spec.html

```diff
@@ -6948,9 +6945,8 @@ <h1>InitializeBinding ( _N_, _V_ )</h1>
 
         <emu-clause id="sec-global-environment-records-setmutablebinding-n-v-s">
           <h1>SetMutableBinding ( _N_, _V_, _S_ )</h1>
-          <p>The concrete Environment Record method SetMutableBinding for global Environment Records attempts to change the bound value of the current binding of the identifier whose name is the value of the argument _N_ to the value of argument _V_. If the binding is an immutable binding, a *TypeError* is thrown if _S_ is *true*. A property named _N_ normally already exists but if it does not or is not currently writable, error handling is determined by the value of the Boolean argument _S_.</p>
+          <p>The SetMutableBinding concrete method of a global Environment Record _envRec_ takes arguments _N_ (a String), _V_ (an ECMAScript language value), and _S_ (a Boolean). It attempts to change the bound value of the current binding of the identifier whose name is the value of the argument _N_ to the value of argument _V_. If the binding is an immutable binding, a *TypeError* is thrown if _S_ is *true*. A property named _N_ normally already exists but if it does not or is not currently writable, error handling is determined by _S_. It performs the following steps when called:</p>
```

Comment:
```suggestion
          <p>The SetMutableBinding concrete method of a global Environment Record _envRec_ takes arguments _N_ (a String), _V_ (an ECMAScript language value), and _S_ (a Boolean). It attempts to change the bound value of the current binding of the identifier whose name is given by _N_ to _V_. When _S_ is *true* or when the binding is a strict binding, SetMutableBinding may throw. It performs the following steps when called:</p>
```

Reply:
"given by _N_ to _V_" is a bit of a garden path. (It's easier to tie the "to" to "given" rather than all the way back to "change".) One way to avoid it would be something like:
> If _envRec_ has a binding for the identifier whose name is given by _N_, this method  attempts to change its bound value to _V_.

As for the two subsequent sentences (in the status quo), I agree that they should probably be replaced, but I'm not sure about the suggested replacement, as it looks like there are other circumstances under which the method may throw (e.g., the binding exists but is uninitialized, or the global object throws when you call its `[[HasProperty]]` or `[[Set]]`). It would be more accurate (though not very precise) to say that there are various things that can cause this method to throw, and for *some* of them, passing `*false*` to `_S_` will suppress the throw (and cause something else to happen instead).

As an alternative to all of the above, maybe the non-standard part of Env Rec method preambles should basically just link to the corresponding 'declaration' in Table [16](https://tc39.es/ecma262/#table-abstract-methods-of-environment-records), [18](https://tc39.es/ecma262/#table-additional-methods-of-function-environment-records), [20](https://tc39.es/ecma262/#table-additional-methods-of-global-environment-records), or [21](https://tc39.es/ecma262/#table-additional-methods-of-module-environment-records). Then we'd only have to get the wording right in one place.